packages/widget/src/ui/PreviewRoute/make-chain-ids-with-actions.ts

```diff
@@ -34,7 +34,9 @@ export const makeChainIDsWithAction = ({
       );
     });
 
-    const swapAction = actions.find((action) => {
+    const reversedActions = actions.slice().reverse();
+    // Find the last swap action that matches conditions
+    const swapAction = reversedActions.find((action) => {
```

Comment:
Seems like the easiest way to do this, too bad findLast is still pretty new ([came out in like 2023](https://www.w3schools.com/js/js_2023.asp)) and lodash/iterating backwards manually in the array feels like overkill